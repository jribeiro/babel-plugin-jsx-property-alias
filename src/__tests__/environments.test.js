// @flow
import { transform } from 'babel-core';

import plugin from '..';

const code = `
React.createClass({
  render: function () {
    return <div className="bar" testID="appiumSelector">
    Hello Wold!
    </div>;
  }
});
`;

describe('environment', () => {
  it('should be a noop when no environment is set', () => {
    const pluginOptions = {
      includeInEnvironments: ['QA'],
      properties: {
        testID: ['accessibilityLabel', 'accessibilityContent'],
        foo: 'bar',
      },
    };

    // Only run plugins targeting the production env.
    // process.env.BABEL_ENV = 'production';
    const actual = transform(code, {
      babelrc: false,
      plugins: [
        [plugin, pluginOptions],
        '@babel/plugin-syntax-jsx',
      ],
    }).code;

    expect(actual).toMatchSnapshot();
  });

  it('should alias testId prop when ALIAS_ENVIRONMENT=QA', () => {
    const pluginOptions = {
      includeInEnvironments: ['QA'],
      properties: {
        testID: ['accessibilityLabel', 'accessibilityContent'],
        foo: 'bar',
      },
    };
    // set environment

    process.env.ALIAS_ENVIRONMENT = 'QA';
    const actual = transform(code, {
      babelrc: false,
      plugins: [
        [plugin, pluginOptions],
        '@babel/plugin-syntax-jsx',
      ],
    }).code;

    expect(actual).toMatchSnapshot();
  });
});
