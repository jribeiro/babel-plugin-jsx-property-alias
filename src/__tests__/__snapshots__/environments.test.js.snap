// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`environment should alias testId prop when ALIAS_ENVIRONMENT=QA 1`] = `
"React.createClass({
  render: function () {
    return <div className=\\"bar\\" testID=\\"appiumSelector\\" accessibilityLabel=\\"appiumSelector\\" accessibilityContent=\\"appiumSelector\\">
    Hello Wold!
    </div>;
  }
});"
`;

exports[`environment should be a noop when no environment is set 1`] = `
"React.createClass({
  render: function () {
    return <div className=\\"bar\\" testID=\\"appiumSelector\\">
    Hello Wold!
    </div>;
  }
});"
`;
